test/integration/framework/shoot_app.go

```diff
@@ -0,0 +1,552 @@
+			if err != nil {
+				return nil, err
+			}
+			return deploymentToReturn, nil
+		}
+		return nil, err
+	}
+	return deploymentToReturn, nil
+}
+
+// WaitForDeploymentToComplete waits for the shoot to be created
+func (o *GardenerTestOperation) WaitForDeploymentToComplete(ctx context.Context, deployment *appsv1.Deployment, pollInterval, pollTimeout time.Duration) error {
+	if o.TillerConfigurationOptions == nil {
+		return ErrEmptyTillerOptionsConfiguration
+	}
+
+	return wait.PollImmediate(pollInterval, pollTimeout, func() (bool, error) {
+		_, err := o.K8sShootClient.Kubernetes().AppsV1().Deployments(o.TillerConfigurationOptions.Namespace).Get(o.TillerConfigurationOptions.DeploymentName, metav1.GetOptions{})
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				o.ShootGardenerTest.Logger.Infof("waiting for deployment: %s to be created \n", o.TillerConfigurationOptions.DeploymentName)
+				return false, nil
+			}
+			return false, err
+		}
+
+		// When all the conditions for the deployment status are true it means that the shoot was successfully created
+		if deploymentCreationCompleted(&deployment.Status) {
+			o.ShootGardenerTest.Logger.Infof("deploymend created!")
+			return true, nil
+		}
+		o.ShootGardenerTest.Logger.Info("Waiting for deployment to be created")
+		return false, nil
+	})
+}
+
+// WaitUntilPodIsRunning waits until the pod with <podName> is running
+func (o *GardenerTestOperation) WaitUntilPodIsRunning(ctx context.Context, podName, podNamespace string, client kubernetes.Interface, pollInterval, pollTimeout time.Duration) error {
+	return wait.PollImmediate(pollInterval, pollTimeout, func() (bool, error) {
+		var (
+			pod *corev1.Pod
+			err error
+		)
+
+		pod, err = client.GetPod(podNamespace, podName)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				o.ShootGardenerTest.Logger.Infof("Waiting for %s to be ready!!", podName)
+				return false, nil
+			}
+			return false, err
+		}
+
+		if pod.Status.Phase == corev1.PodRunning {
+			o.ShootGardenerTest.Logger.Infof("%s seems to be ready!!", podName)
+			return true, nil
+		}
+
+		o.ShootGardenerTest.Logger.Infof("Waiting for %s to be ready!!", podName)
+		return false, nil
+	})
+}
+
+// WaitUntilPodWithLabelsIsReady wait until pod with labels <podLabels> is running
+func (o *GardenerTestOperation) WaitUntilPodWithLabelsIsReady(ctx context.Context, podLabels map[string]string, namespace string, client kubernetes.Interface, pollInterval, pollTimeout time.Duration) error {
+	return wait.PollImmediate(pollInterval, pollTimeout, func() (bool, error) {
+		var (
+			pod      *corev1.Pod
+			podReady bool
+			err      error
+		)
+
+		pod, err = o.getPodByLabels(ctx, podLabels, namespace, client)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				o.ShootGardenerTest.Logger.Infof("Waiting for pod with labels: %v to be ready!!", podLabels)
+				return false, nil
+			}
+			return false, err
+		}
+
+		for _, condition := range pod.Status.Conditions {
```

Comment:
Didn't find it? is it in "health"?

Reply:
https://github.com/gardener/gardener/blob/master/pkg/utils/kubernetes/health/health.go#L83:6 - please check with @adracus whether to use it here or not

Reply:
Ok, I will change the implementation to look at deployments instead. But worth noticing that `health.CheckObject` only gives an error so I am not able to differentiate if the error occured because of an unhandled error in the inner implementation of `CheckDeployment` or because it just found the status to be false.